x-pack/test/functional/services/ml/anomaly_charts.ts

```diff
@@ -13,20 +13,32 @@ export type MlAnomalyCharts = ProvidedType<typeof AnomalyChartsProvider>;
 
 export function AnomalyChartsProvider({ getService }: FtrProviderContext) {
   const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const find = getService('find');
 
   return {
     async assertAnomalyExplorerChartsCount(
-      chartsContainerSubj: string,
+      chartsContainerSubj: string | undefined,
       expectedChartsCount: number
     ) {
-      const chartsContainer = await testSubjects.find(chartsContainerSubj);
-      const actualChartsCount = (
-        await chartsContainer.findAllByClassName('ml-explorer-chart-container', 3000)
-      ).length;
-      expect(actualChartsCount).to.eql(
-        expectedChartsCount,
-        `Expect ${expectedChartsCount} charts to appear, got ${actualChartsCount}`
-      );
+      await retry.tryForTime(5000, async () => {
+        // For anomaly charts, time range expected is of the chart plotEarliest/plotLatest
+        // and not of the global time range
+        // but since plot earliest & latest might vary depends on the current time
+        // we don't know the exact hashed id for sure
+        // so we find first available chart container if id is not provided
+        const chartsContainer =
+          chartsContainerSubj !== undefined
+            ? await testSubjects.find(chartsContainerSubj)
+            : await find.byCssSelector('[class="mlAnomalyExplorerEmbeddableWrapper"]');
```

Comment:
I reckon functional tests should not rely on CSS classes. Instead, you can set another data-test-subj to an element. There is an `ExplorerChartContainer` component where you can add it.

Reply:
Updated here [`e7b6a11` (#140147)](https://github.com/elastic/kibana/pull/140147/commits/e7b6a11494990f76c893c8b15b701b5697121ca0)